src/Microsoft.TemplateEngine.Edge/Settings/TemplateInfo.cs

```diff
@@ -108,7 +109,12 @@ internal TemplateInfo(ITemplate template, ILocalizationLocator? localizationInfo
                 logger.LogDebug($"Start loading host config {HostConfigPlace}");
                 try
                 {
-                    using (var sr = new StreamReader(template.TemplateSourceRoot.FileInfo(HostConfigPlace).OpenRead()))
+                    IFile? hostFile = template.TemplateSourceRoot?.FileInfo(HostConfigPlace);
+                    if (hostFile == null || !hostFile.Exists)
+                    {
+                        throw new Exception($"Host file {hostFile} does not exist.");
```

Comment:
`IFile` implementations do not override `ToString` - either override it, or output some more usefull information here